go/vt/vtgate/planbuilder/ddl.go

```diff
@@ -132,3 +114,106 @@ func buildDDLPlans(sql string, ddlStatement sqlparser.DDLStatement, vschema Cont
+	selectPlan, err = createInstructionFor(sqlparser.String(ddl.Select), ddl.Select, vschema)
+	if err != nil {
+		return nil, nil, err
+	}
+	routePlan, isRoute := selectPlan.(*engine.Route)
+	if !isRoute {
+		return nil, nil, vterrors.New(vtrpc.Code_INVALID_ARGUMENT, ViewComplex)
+	}
+	if keyspace.Name != routePlan.GetKeyspaceName() {
+		return nil, nil, vterrors.New(vtrpc.Code_INVALID_ARGUMENT, ViewDifferentKeyspace)
+	}
+	if routePlan.Opcode != engine.SelectUnsharded && routePlan.Opcode != engine.SelectEqualUnique && routePlan.Opcode != engine.SelectScatter {
+		return nil, nil, vterrors.New(vtrpc.Code_INVALID_ARGUMENT, ViewComplex)
+	}
+	sqlparser.Rewrite(ddl.Select, func(cursor *sqlparser.Cursor) bool {
+		switch tableName := cursor.Node().(type) {
+		case sqlparser.TableName:
+			cursor.Replace(sqlparser.TableName{
+				Name: tableName.Name,
+			})
+		}
+		return true
+	}, nil)
+	return destination, keyspace, nil
+}
+
+func buildCreateView(vschema ContextVSchema, ddl *sqlparser.CreateView) (key.Destination, *vindexes.Keyspace, error) {
+	// For Create View, we require that the keyspace exist and the select query can be satisfied within the keyspace itself
+	// We should remove the keyspace name from the table name, as the database name in MySQL might be different than the keyspace name
+	destination, keyspace, _, err := vschema.TargetDestination(ddl.ViewName.Qualifier.String())
+	if err != nil {
+		return nil, nil, err
+	}
+	ddl.ViewName.Qualifier = sqlparser.NewTableIdent("")
+
+	var selectPlan engine.Primitive
+	selectPlan, err = createInstructionFor(sqlparser.String(ddl.Select), ddl.Select, vschema)
+	if err != nil {
+		return nil, nil, err
+	}
+	routePlan, isRoute := selectPlan.(*engine.Route)
+	if !isRoute {
+		return nil, nil, vterrors.New(vtrpc.Code_INVALID_ARGUMENT, ViewComplex)
+	}
+	if keyspace.Name != routePlan.GetKeyspaceName() {
+		return nil, nil, vterrors.New(vtrpc.Code_INVALID_ARGUMENT, ViewDifferentKeyspace)
+	}
+	if routePlan.Opcode != engine.SelectUnsharded && routePlan.Opcode != engine.SelectEqualUnique && routePlan.Opcode != engine.SelectScatter {
+		return nil, nil, vterrors.New(vtrpc.Code_INVALID_ARGUMENT, ViewComplex)
+	}
+	sqlparser.Rewrite(ddl.Select, func(cursor *sqlparser.Cursor) bool {
+		switch tableName := cursor.Node().(type) {
+		case sqlparser.TableName:
+			cursor.Replace(sqlparser.TableName{
+				Name: tableName.Name,
+			})
+		}
+		return true
+	}, nil)
+	return destination, keyspace, nil
+}
+
+func buildDropViewOrTable(vschema ContextVSchema, ddlStatement sqlparser.DDLStatement) (key.Destination, *vindexes.Keyspace, error) {
+	var destination key.Destination
+	var keyspace *vindexes.Keyspace
+	for i, tab := range ddlStatement.GetFromTables() {
+		table, _, _, _, destinationTab, err := vschema.FindTableOrVindex(tab)
+		if err != nil {
+			return nil, nil, err
+		}
+		if table == nil {
+			return nil, nil, vterrors.Errorf(vtrpc.Code_INVALID_ARGUMENT, "table does not exists: %s", tab.Name.String())
+		}
```

Comment:
I think we should not assume the the table will be present in vschema.
Also, for unsharded keyspace there is no need to provide tables in the vschema.

Reply:
For sharded keyspace we want the table to be present in the vschema. For unsharded keyspaces it does not require that the table be present in the vschema.  There is a check for that in the implementation of the function. I have also added a test for the same.

Reply:
As per discussion, will make the necessary changes

Reply:
done in the last commit